src/common/config.cpp

```diff
@@ -128,12 +128,14 @@ bool ArgsManager::ReadConfigFiles(std::string& error, bool ignore_invalid_keys)
     }
 
     const auto conf_path{GetConfigFilePath()};
-    std::ifstream stream{conf_path};
-
-    // not ok to have a config file specified that cannot be opened
-    if (IsArgSet("-conf") && !stream.good()) {
-        error = strprintf("specified config file \"%s\" could not be opened.", fs::PathToString(conf_path));
-        return false;
+    std::ifstream stream;
+    if (!conf_path.empty()) { // path is empty when -noconf is specified
```

Comment:
This bothers me more than it should: we're adding comments instead of solving these problems with code - conf_path should be an optional in that case instead of coming up with special value

Reply:
We are currently using the nullness of `m_config_path` to signal something different, initialized/not, although I think it should probably be more explicit. See https://github.com/bitcoin/bitcoin/pull/31212#issuecomment-2470510859, and this code:

https://github.com/bitcoin/bitcoin/blob/2ac863bdf974d7b4b4e8314876d02270400b6240/src/common/args.cpp#L761-L772

We could possibly do `optional<optional<path>> m_config_path` to flag both initialization and negation... but I'm not sure I want to go down that road. Do you have any concrete suggestion?

Reply:
optional of optional sounds like a `std::variant` - but I'm fine with doing these in a separate PR, of course